api/src/types/assets.ts

```diff
@@ -65,21 +65,11 @@ export type TransformationMap = {
 
 export type Transformation = TransformationMap[keyof TransformationMap];
 
+export type TransformationResize = Pick<ResizeOptions, 'width' | 'height' | 'fit' | 'withoutEnlargement'>;
+
 export type TransformationParams = {
 	key?: string;
 	transforms?: Transformation[];
-};
-
-// Transformation preset is defined in the admin UI.
-export type TransformationPreset = TransformationPresetFormat &
-	TransformationPresetResize &
-	TransformationParams & { key: string };
-
-export type TransformationPresetFormat = {
-	format?: 'jpg' | 'jpeg' | 'png' | 'webp' | 'tiff' | 'avif';
+	format?: 'auto' | 'jpg' | 'jpeg' | 'png' | 'webp' | 'tiff' | 'avif';
```

Comment:
Small note on the side: While `auto` is now a valid input for presets on the user side, it must not be passed to `resolvePreset()`